npm/hermes-engine-darwin/unpack-tarball.js

```diff
@@ -0,0 +1,19 @@
+const path = require("path");
+const fs = require("fs");
+const child_process = require("child_process");
+
+if (
+  process.platform === "darwin" &&
+  !fs.existsSync(path.join(__dirname, "destroot"))
+) {
+  const tarball = fs.readdirSync(__dirname).find(function (entry) {
+    return entry.endsWith(".tar.gz");
+  });
+  if (!tarball) {
+    throw new Error("Could not locate tarball");
+  }
+  child_process.execFileSync("/usr/bin/tar", [
```

Comment:
Why an explicit path?  Some people might install tools in /usr/local/bin, for example.  If you just say "tar", will node search the path?

Reply:
The system supplied version of `tar` is a known good version, so it seems more stable to _not_ accidentally use some other version of `tar` that the user might have installed.

To your second question, seems like Node will indeed search the `PATH` env variable when not give an absolute path.

Reply:
@alloy I definitely see your point, though if a user explicitly wants to supply their own `tar` I think we should let them.

Our CircleCI jobs make some use of this kind of thing, and my `~/bin` has several overrides that I want used and therefore put first in my PATH.

Reply:
I personally find it a little less intuitive for an opaque installation, such as untarring from inside a post-install step, to possibly break due to such customizabitility, but on the other hand I don’t really expect people to install a custom tar without knowing what they’re doing, so will make the change 👍

Reply:
Done